net-p2p/rtorrent/files/rtorrent-0.10.0-lto.patch

```diff
@@ -0,0 +1,89 @@
+https://bugs.gentoo.org/861848
+https://github.com/rakshasa/rtorrent/issues/1264
+https://github.com/rakshasa/rtorrent/pull/1291
+https://github.com/rakshasa/rtorrent/commit/c1d076f1822752d4956a884a9aa9e6b0026b11f3
+https://github.com/rakshasa/rtorrent/commit/38b39bdafc9edd7e7e72672e6fcbd397b6c2cab8
+
+From c1d076f1822752d4956a884a9aa9e6b0026b11f3 Mon Sep 17 00:00:00 2001
+From: stickz <stickman002@mail.com>
+Date: Sun, 29 Sep 2024 13:24:01 -0400
+Subject: [PATCH] Fixing static building
```

Comment:
This patch has nothing to do with LTO. It is a totally unrelated issue and should not have metadata in the patch header (or filename) talking about LTO.

You may link to https://github.com/rakshasa/rtorrent/pull/1296 instead...

Reply:
OK, I see. Thank for the clarification.

Reply:
I am not sure what exactly the patch actually does, to be quite honest.

It causes /usr/bin/rtorrent to be compiled using libtool instead of `x86_64-pc-linux-gnu-g++` directly:'


```
# without
x86_64-pc-linux-gnu-g++ -std=c++14  -pipe -march=native -fstack-protector-all -O2 -fdiagnostics-color=always -frecord-gcc-switches -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wformat -Werror=format-security -DNDEBUG -Wall -isysroot / -pthread     -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -I/usr/include/ncursesw  -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,--sysroot,/ -o rtorrent main.o libsub_root.a -lpthread -lpthread  -lncursesw -ltinfow -lcppunit  -lcurl -ltorrent 


# with
libtool: link: x86_64-pc-linux-gnu-g++ -std=c++14 -pipe -march=native -fstack-protector-all -O2 -fdiagnostics-color=always -frecord-gcc-switches -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wformat -Werror=format-security -DNDEBUG -Wall -isysroot / -D_DEFAULT_SOURCE -D_XOPEN_SOURCE=600 -I/usr/include/ncursesw -Wl,-O1 -Wl,-z -Wl,pack-relative-relocs -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wl,--defsym=__gentoo_check_ldflags__=0 -Wl,--sysroot -Wl,/ -o rtorrent main.o  -Wl,--as-needed libsub_root.a -lpthread -lncursesw -ltinfow -lcppunit -lcurl -ltorrent -pthread
```

libtool should not, in principle, be necessary for statically linking at all.

Reply:
Ok. I remove this **non**-lto.

Reply:
Sure.